lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -3423,6 +3423,20 @@ Value FIRRTLLowering::createArrayIndexing(Value array, Value index) {
   auto size = hw::type_cast<hw::ArrayType>(array.getType()).getSize();
   auto valWire = builder.create<sv::WireOp>(
       hw::type_cast<hw::ArrayType>(array.getType()).getElementType());
+
+  // Extend to power of 2
+  if (!llvm::isPowerOf2_64(size)) {
+    auto extElem =
+        builder.create<hw::ConstantOp>(APInt(llvm::Log2_64_Ceil(size), 0));
+    auto extValue = builder.create<hw::ArrayGetOp>(array, extElem);
+    SmallVector<Value> temp(llvm::NextPowerOf2(size) - size, extValue);
+    auto ext = builder.create<hw::ArrayCreateOp>(temp);
+    SmallVector<Value> temp2;
+    temp2.push_back(ext.getResult());
+    temp2.push_back(array);
+    array = builder.create<hw::ArrayConcatOp>(temp2);
```

Comment:
Golf nit: Does this work?

```suggestion
    array = builder.create<hw::ArrayConcatOp>({ext.getResult(), array});
```

Reply:
no, at least when I tried earlier.